feat/tx-comps: Add RecipientSelector component

emizzle opened the pull request:
Fixes #671.

Based on the spec in https://www.notion.so/emizzle/Wallet-transaction-components-2003b78a8d0d41c4ab3d21eb2496fb20, this component handles user input for a recipient address, which can be sourced from manual address input, ENS name, contact selection, or another of the user's wallet accounts.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9c3dc99f | [#1](https://ci.status.im/job/nim-status-client/job/macos/job/PR-719/1/) | 2020-08-10 07:33:10 | ~9 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-200810-072405-9c3dc9-pr.dmg) |

PascalPrecht commented on ui/shared/ContactSelector.qml:
```diff
@@ -0,0 +1,186 @@
+                Text {
+                    text: name
+                    font.pixelSize: 15
+                    font.family: Style.current.fontBold.name
```
I noticed this one is extremely "bold", but there also doesn't seem to be an actual design for that..

jrainville replied:
I didn't look at the design nor what this actually looks like right now, but most of the time in the design, we just use `fonttWeight: medium`

emizzle replied:
The design looks like this: 
![Imgur](https://imgur.com/ejIf2GR.png)
Have updated this to use `fontRegular.name` and `font.bold: true` instead.

jrainville commented on ui/shared/AddressInput.qml:
```diff
@@ -0,0 +1,89 @@
+    function isValidAddress(inputValue) {
+        return /0x[a-fA-F0-9]{40}/.test(inputValue)
+    }
+    function isValidEns(inputValue) {
```
I'd put those two functions in `Utils.qml`

emizzle replied:
`isValidEns` needs to change at some point, and it will not go in `Utils`, but I have moved the address regex to Utils.

emizzle replied:
Also, note I updated the regex in `Utils.isHex` to include uppercase characters as well. Please let me know if this was an incorrect assumption.

jrainville replied:
Sounds good

emizzle commented:
Thanks for the reviews @PascalPrecht and @jrainville! I've updated the PR with the requested changes. To note, I've also added in the ability for the `RecipientSelector` to be in `readOnly` mode (as per the requirements for chat commands), so it might need another look.

jrainville commented on ui/shared/ContactSelector.qml:
```diff
@@ -0,0 +1,187 @@
+        } else {
+            select.select.border.color = Style.current.transparent
+            select.select.border.width = 0
+            validationErrorText.visible = false
```
Note, for the Input, I created a validationError prop. Maybe it's worth adding it to the Select as well so that it handles its borders itself.

emizzle replied:
Good idea, updated the Select component to handle the validation!
